src/hotspot/share/opto/constantTable.hpp

```diff
@@ -38,18 +38,21 @@ class ConstantTable {
   class Constant {
   private:
     BasicType _type;
+    bool _is_array;
```

Comment:
Can you rely on `_type` and just use `T_ARRAY` tag for that purpose?

Reply:
In that case, an additional field would be needed to identify the type of the elements anyway.